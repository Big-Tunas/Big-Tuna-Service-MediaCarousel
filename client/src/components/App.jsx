import React from 'react';
import axios from 'axios';
import MediaCarousel from './MediaCarousel.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      media: []
    };
  }

  componentDidMount() {
    this.getMediaDataFromServer();
  }

  getMediaDataFromServer() {
    axios.get('/mediaData')
      .then((response) => {
        this.setState({ media: response.data })
      })
      .catch((error) => {
        console.log(error);
      })
  };

  render() {
    return (
      <div>
        <MediaCarousel data={this.state.media}/>
      </div>
    )
  }
};

export default App;
