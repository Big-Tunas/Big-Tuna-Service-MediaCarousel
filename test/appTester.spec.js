
// ========================== SAMPLE TESTER FILE ===============================================

import React from 'react';
import renderer from 'react-test-renderer';
import App, { Counter } from '../client/src/components/appTester.jsx';

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Counterzz', () => {
  test('snapshot renders!!!!', () => {
    const component = renderer.create(<Counter counter={1} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


// need test in appTester.jsx

// import React from 'react';
// import { shallow } from 'enzyme';
// import MyComponent from '../client/src/components/appTester.jsx';

// describe('MyComponent', () => {
//   it('should render correctly in "debug" mode', () => {
//     const component = shallow(<MyComponent debug />);

//     expect(component).toMatchSnapshot();
//   });
// });