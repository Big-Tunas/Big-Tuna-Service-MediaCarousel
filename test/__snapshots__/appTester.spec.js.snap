// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div>
  <h1>
    My Counter
  </h1>
  <div>
    <p>
      0
    </p>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Increment
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Decrement
  </button>
  <h2>
    My Async Data
  </h2>
  <ul />
</div>
`;

exports[`Counterzz snapshot renders!!!! 1`] = `
<div>
  <p>
    1
  </p>
</div>
`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
